profile.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash
+
+function profile() {
+    if [ ! -d ~/FlameGraph ]; then
+        InstallFlameGraph
+    fi
+
+    case "$OSTYPE" in
+        darwin*)
+            profileMac "$1"
+            ;;
+        linux-gnu*|msys)
+            profileLinux "$1"
+            ;;
+        *)
+            echo "Error: Unknown OS"
+            exit 1
+            ;;
+    esac
+}
+
+function profileLinux() {
+    local mojo_file=$1
+    LinuxInstallDependencies
+    LinuxPermissions
+    runProfile "$mojo_file"
+}
+
+function profileMac() {
+    local mojo_file=$1
+    MacInstallDependencies
+    MacPermissions
+    runProfile "$mojo_file"
+}
+
+function runProfile() {
+    local mojo_file=$1
+    local mojo_name="${mojo_file%.mojo}"
+    local temp_dir="./temp"
+    local perf_output="$temp_dir/out.perf"
+    local flamegraph_output="flamegraph.svg"
+
+    echo "Profiling $mojo_file..."
+
+    mkdir -p "$temp_dir"
+
+    echo "Building $mojo_file..."
+    mojo build -O0 -I . "$mojo_file"
+
+    echo "Stripping debug symbols..."
+    mv "$mojo_name" "$temp_dir/run.exe"
+    llvm-strip --strip-debug "$temp_dir/run.exe"
+
+    echo "Running perf record..."
+    perf record -F 99 -a -g -o "$perf_output" -- "$temp_dir/run.exe"
```

Comment:
For some reason i still need a sudo here to get it to work. 
And does it work for you without the .exe (=windows specific)